test/addons/async-hooks-promise/test.js

```diff
@@ -11,16 +11,18 @@ if (process.env.NODE_TEST_WITH_ASYNC_HOOKS) {
 }
 
 // Baseline to make sure the internal field isn't being set.
+// Because of bootstrap tracking this will be enabled.
 assert.strictEqual(
-  binding.getPromiseField(Promise.resolve(1)),
-  0);
+  binding.getPromiseField(Promise.resolve(1)).isChainedPromise,
+  undefined);
```

Comment:
Why is this updated? `(0).isChainedPromise` is still undefined, so this looks like potentially hiding a flake. I am not sure if `isChainedPromise` is even a valid value for the promise wraps.

Reply:
This is set when async_hooks is enabled, but for some reason it is not set with this refactoring PR and I'm not sure why. The check is specifically checking though that `isChainedPromise` is falsy so the adjustment still aligns with the intent.

Any suggestions why this changed are welcome... still can't see where it's happening myself.

Reply:
I suspect it is due to very subtle promise timings, but the point is that it is a non-breaking change. 

@jasnell if you could confirm this test change is non-breaking for async hooks that would be a huge help (timings seem to be altered somehow in that this field doesn't yet exist, instead of existing and being disabled, but all other tests pass).

Reply:
Ok sorry for confusion, worked this one out - I wasn't reverting the test case itself properly after the previous work. All is good here with the same test as always.